packages/dapp/src/redux/store.ts

```diff
@@ -2,14 +2,31 @@ import thunk from "redux-thunk";
 import { createStore, compose, applyMiddleware } from "redux";
 import { routerMiddleware } from "connected-react-router";
 
-import { gaMiddleware } from "./analytics";
 import { createBrowserHistory } from "history";
 import createRootReducer from "./reducers";
+import dynamicMiddlewares from "redux-dynamic-middlewares";
+import cookie from "react-cookies";
+import { gaMiddleware } from "./analytics";
 
 export const history = createBrowserHistory();
 
-export const store = createStore(
-  createRootReducer(history),
-  {},
-  compose(applyMiddleware(routerMiddleware(history), thunk, gaMiddleware)),
-);
+export const store = () => {
+  const didAnswerTracking = cookie.load("didAnswerTracking");
+  if (didAnswerTracking) {
+    const allowTracking = cookie.load("allowTracking");
+    if (allowTracking) {
+      console.log("THIS ONE");
+      return createStore(
+        createRootReducer(history),
+        {},
+        compose(applyMiddleware(routerMiddleware(history), thunk, gaMiddleware)),
```

Comment:
if this middleware isnt there what does trackpageview do?

Reply:
nothing, when the middleware isn't there no actions are tracked. when the middleware is added it picks them up again.